### ISSUE TITLE
Bug: Incorrect return type from TSESLint.FlatESLint#calculateConfigForFile

### ISSUE THREAD

**Issue by Gajdascz** [typescript-eslint/typescript-eslint#11444]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

tsconfig-utils

### Playground Link

https://stackblitz.com/edit/typescript-eslint-utils-type-err?file=index.cjs

### Repro Code

```TypeScript
import { TSESLint } from '@typescript-eslint/utils';

const eslint = new TSESLint.FlatESLint({
  overrideConfigFile: './eslint.config.js',
});

// The type here is FlatConfig.ConfigArray,
// but runtime result is a merged config object, not an array
const config = await eslint.calculateConfigForFile('index.ts');

console.log(Array.isArray(config)); // false
console.log(typeof config);        // object
console.log(Object.keys(config));  // shows config fields, not array elements
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc

```

### Expected Result

The return type of TSESLint.FlatESLint#calculateConfigForFile() should be:

`Promise<FlatConfig.Config>`

This matches the [ESLint Node.js API docs](https://eslint.org/docs/latest/integrate/nodejs-api#-eslintcalculateconfigforfilefilepath), which state that this method returns a resolved configuration object, not a config array.

### Actual Result

Currently, the method is typed as returning:

Promise<FlatConfig.ConfigArray>

See source:
https://github.com/typescript-eslint/typescript-eslint/blob/12e6961429c87ff80eb79ea34daac8bb73ee7be3/packages/utils/src/ts-eslint/eslint/FlatESLint.ts#L14-L21

This is incorrect, as calculateConfigForFile() returns a merged config object, not a config array.

### Additional Info

This applies to all TSESLint ESLint wrapper classes in @typescript-eslint/utils, since they just alias a common base

https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/utils/src/ts-eslint/ESLint.ts

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/utils`             | `8.38.0` |
| `TypeScript`                           | `5.9.2` |
| `ESLint`                               | `9.32.0` |
| `node`                                 | `22.14.0` |
| `typescript-eslint`                    | `8.38.0` |

**Comment by kirkwaiblinger** [comment]
Hi @Gajdascz!

Yep, looks like a mistake. Feel free to send a PR 🙂 

Thanks for the nice, easy-to-use demo code!

**Comment by thebolarin** [comment]
@kirkwaiblinger Can i pick this up?
